src/aatoolbox/datasources/glofas/glofas.py

```diff
@@ -0,0 +1,453 @@
+            is_public=True,
+        )
+        # The GloFAS API on CDS requires coordinates have the format x.x5
+        self._geo_bounding_box = geo_bounding_box.round_coords(
+            offset_val=0.05, round_val=0.1
+        )
+        self._start_date = start_date
+        self._end_date = end_date
+        self._cds_name = cds_name
+        self._system_version = system_version
+        self._product_type = product_type
+        self._date_variable_prefix = date_variable_prefix
+        self._frequency = frequency
+        self._coord_names = coord_names
+        self._leadtime_max = leadtime_max
+        self._forecast_type = type(self).__name__.replace("Glofas", "").lower()
+        self._date_range = rrule.rrule(
+            freq=self._frequency,
+            dtstart=self._start_date,
+            until=self._end_date,
+        )
+
+    def download(  # type: ignore
+        self,
+        clobber: bool = False,
+    ) -> List[Path]:
+        """
+        Download the GloFAS data by querying CDS.
+
+        Parameters
+        ----------
+        clobber : bool, default = False
+            Overwrite files that were already downloaded
+
+        Returns
+        -------
+        A list paths of downloaded files
+
+        """
+        msg = (
+            f"Downloading GloFAS {self._forecast_type} "
+            f"for {self._start_date} - {self._end_date}"
+        )
+        if self._leadtime_max is not None:
+            msg += f"and up to {self._leadtime_max} day lead time"
+        logger.info(msg)
+
+        # Make directory
+        output_directory = self._get_directory()
+        output_directory.mkdir(parents=True, exist_ok=True)
+
+        # Get list of files to open
+        query_params_list = []
+        for date in self._date_range:
+            output_filepath = self._get_filepath(
+                year=date.year,
+                month=date.month,
+                day=date.day,
+                leadtime_max=self._leadtime_max,
+            )
+            if not clobber and output_filepath.exists():
+                continue
+            query_params_list.append(
+                _QueryParams(
+                    filepath=output_filepath,
+                    query=self._get_query(
+                        year=date.year,
+                        month=date.month,
+                        day=date.day,
+                        leadtime_max=self._leadtime_max,
+                    ),
+                )
+            )
+
+        download_filepaths = self._download(
+            query_params_list=query_params_list
+        )
+        logger.info(
+            f"Downloaded {len(download_filepaths)} files to {output_directory}"
+        )
+        logger.debug(f"Files downloaded: {download_filepaths}")
+        return download_filepaths
+
+    def process(  # type: ignore
+        self,
+        clobber: bool = False,
+    ) -> List[Path]:
+        """
+        Process the downloaded GloFAS files.
```

Comment:
I think some more details here would help. While the overall walkthrough of GloFAS is clear, I think having function level documentation on what the steps are doing might also be helpful. What are we processing, what for? Or do you think we just leave documentation primarily in the overall module guidance?

Reply:
Oops you're right, I was being lazy, I will add some more info to download & process